### ISSUE TITLE
Add ability to opt out of IP auditing

### ISSUE THREAD

**Issue by jmduke** [buttondown-email/roadmap#1967]
## TL;DR

If Buttondown detects a problematic/spammy IP address, we remove all subscribers and block incoming ones for that address. Some authors would like to disable this; sometimes the IP address blocklist can be aggressive. 

## Current flow

![image](https://github.com/buttondown-email/roadmap/assets/461046/3532694b-f707-4536-b2a6-21bab436d58e)

## Background notes

- The most important model in the world is pretty much the **Newsletter** model. That's where all the settings lie (look at how many fields it has!) and where you'll be adding this new setting. 
- To make new settings available to the frontend (so that users can update/save them), you'll want to modify the `NewsletterSerializer`. "Serializers" are a DRF concept that aren't worth worrying too much about — the only real reason we're using a serializer here (as opposed to `django-ninja`, which is used almost everywhere else) is out of laziness, since this is fairly old code.
- The other important model for this task is the `ProhibitedIPAddress` model, which... is exactly what it sounds like. You shouldn't need to worry too much about how it's being created, since that logic doesn't need to change — rather, you need to modify what happens *after* it's been created, which exists in a signal titled `prohibited_ip_address_creation_actions`. (I am not a huge fan of signals because they are hard to locate and lead to scattered code; this one was written almost four years ago.)

## Suggested breakdown

- [ ] Add an inert flag to the Newsletter model, `Newsletter.should_enable_ip_auditing`. Make sure it's visible in the admin as well!
- [ ]  Add logic to `prohibited_ip_address_creation_actions` to avoid Newsletters that have opted out of IP-based auditing. (Be sure to add a test!)
- [ ]  Add logic to `prohibited_ips.Middleware` to not throw a 404 if the newsletter being accessed has opted out of IP-based auditing. (Be sure to add a test!)
- [ ] Add front-end logic to the settings page (under "Subscription Flow") to allow folks to toggle in and out of IP-based auditing.
